aptos-move/e2e-move-tests/src/tests/function_value_capture_option.rs

```diff
@@ -0,0 +1,367 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Tests for capturing option type in function values.
+
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_framework::BuildOptions;
+use aptos_package_builder::PackageBuilder;
+use aptos_transaction_simulation::Account;
+use aptos_types::{
+    on_chain_config::FeatureFlag,
+    transaction::{ExecutionStatus, TransactionStatus},
+};
+use claims::assert_matches;
+use move_core_types::account_address::AccountAddress;
+
+#[test]
+fn test_vm_value_fv_capture_option_1() {
+    let mut h = MoveHarness::new();
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0x99").unwrap());
+
+    // test directly capture option type
+    h.enable_features(vec![], vec![FeatureFlag::ENABLE_CAPTURE_OPTION]);
+    let status = publish(
+        &mut h,
+        &acc,
+        r#"
+        module 0x99::m {
+            use std::option;
+
+            struct FunctionStore has key {
+                f: ||option::Option<u64> has copy+drop+store,
+            }
+
+            public fun id(x: option::Option<u64>): option::Option<u64> {
+                x
+            }
+
+            fun init_module(account: &signer) {
+                let v = option::none();
+                let f: ||option::Option<u64> has copy+drop+store = || id(v);
+                move_to(account, FunctionStore { f });
+            }
+        }
+        "#,
+    );
+    assert_matches!(
+        status,
+        TransactionStatus::Keep(ExecutionStatus::ExecutionFailure { .. })
```

Comment:
So the error status is not propagated? This might be quite bad for users unless we have a compiler warning @rahxephon89 @wrwg @vineethk

Reply:
Yeah, this quite bothers me. Do you know why it is the case? Does the logic of propagation rely on the range of error code?

Reply:
I think so, we can maybe change it to verification code iirc this one is kept as `Miscellaneous(status_code)`